packages/medusa/src/services/cart.ts

```diff
@@ -1385,14 +1389,14 @@ class CartService extends TransactionBaseService<CartService> {
           )
         }
 
-        const session = await this.paymentProviderService_
+        const session = (await this.paymentProviderService_
           .withTransaction(transactionManager)
-          .authorizePayment(cart.payment_session, context)
+          .authorizePayment(cart.payment_session, context)) as PaymentSession
```

Comment:
there seems to be a typing mismatch here since `authorizePayment` in the interface doesn't return a PaymentSession

Reply:
here it is the payment-provider service which return a payment session, it is not the plugin, which is called within the payment provider service